nova/virt/vmwareapi/cluster_util.py

```diff
@@ -190,43 +211,57 @@ def update_placement(session, cluster, vm_ref, group_infos):
     reconfigure_cluster(session, cluster, config_spec)
 
 
-def _create_cluster_rules_spec(client_factory, name, vm_refs,
-                               policy='affinity', operation="add",
-                               rule=None):
+def create_vm_rule(client_factory, name, vm_refs, policy='affinity',
+                   rule=None):
+    """Create a ClusterAffinityRuleSpec or ClusterAntiAffinityRuleSpec object
 
-    rules_spec = client_factory.create('ns0:ClusterRuleSpec')
-    rules_spec.operation = operation
-    if policy == 'affinity' or policy == 'soft-affinity':
-        policy_class = 'ns0:ClusterAffinityRuleSpec'
-    elif policy == 'anti-affinity' or policy == 'soft-anti-affinity':
-        policy_class = 'ns0:ClusterAntiAffinityRuleSpec'
-    else:
-        msg = _('%s policy is not supported.') % policy
-        raise exception.Invalid(msg)
+    :param:policy: Defines with of the object types is created. "affinity" and
+                   "soft-affinity" map to ClusterAffinityRuleSpec,
+                   "anti-affinity" and "soft-anti-affinity" map to
+                   ClusterAntiAffinityRuleSpec. Ignored if "rule" is given
+    :param:rule: if given, don't create any object, but instead update the
+                 given object's attributes. "policy" is ignored here.
+    """
+    if rule is None:
+        if policy in ('affinity', 'soft-affinity'):
+            obj_type = 'ns0:ClusterAffinityRuleSpec'
+        elif policy in ('anti-affinity', 'soft-anti-affinity'):
+            obj_type = 'ns0:ClusterAntiAffinityRuleSpec'
+        else:
+            msg = 'Policy {} is not supported.'.format(policy)
+            raise exception.ValidationError(msg)
```

Comment:
Wouldn't the standard-lib exception [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) be more appropriate here?